src/test/java/org/apache/geronimo/microprofile/impl/jwtauth/jwt/KidMapperTest.java

```diff
@@ -0,0 +1,52 @@
+package org.apache.geronimo.microprofile.impl.jwtauth.jwt;
+
+import static javax.ws.rs.client.ClientBuilder.newClient;
+import static org.testng.Assert.assertEquals;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.arquillian.testng.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+public class KidMapperTest extends Arquillian {
+
+    private static JwksServer jwksServer;
+
+    @Deployment()
+    public static WebArchive createDeployment() throws Exception {
+        jwksServer = new JwksServer();
+        jwksServer.start();
+        System.setProperty("mp.jwt.verify.publickey.location", "http://localhost:" + jwksServer.getPort() + "/jwks.json");
```

Comment:
where is it cleaned?

tip: inject it in the war as a mp properties file

side note: true for all tests